src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,281 @@
+        sum = op1 * op2
+        return {'oper': '*', 'op2': op2, 'sum': sum}
+
+    def div(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
+        sum = op1 / op2
+        if sum > 0:
+            sum = int(math.floor(sum))
+        else :
+            sum = int(math.ceil(sum))
+        return {'oper': '/', 'op2': op2, 'sum': sum}
+
+    def mod(self, op1, op2):
+        if op2 == 0:
+            op2 = 2
+        if (((op1 < 0 and op2 > 0) or (op1 > 0 and op2 < 0)) and (op1 % op2 != 0)):
+            sum = ((op1 % op2) - op2)
+        else:
+            sum = (op1 % op2)
+        return {'oper': '%', 'op2': op2, 'sum': sum}
+
+    def bitshiftleft(self, op1, op2):
+        op2 = int(math.fabs(op2))
+        if op1 < 0:
+            op2 = math.floor(math.log(MIN/op1))
+        else:
+            op2 = math.ceil(math.log(MIN/op1))
+        sum = op1 << op2
+        if sum < MIN:
+            if op1 < 0:
+                op2 = math.floor(math.log(MIN/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MIN/op1, 2))
+        elif sum > MAX:
+            if op1 > 0:
+                op2 = math.floor(math.log(MAX/op1, 2))
+            else:
+                op2 = math.ceil(math.log(MAX/op1, 2))
+        sum = op1 << op2
+        return {'oper': '<<', 'op2': op2, 'sum': sum}
+
+    def bitand(self, op1, op2):
+        sum = op1 & op2
+        if sum < MIN:
+            op2 = sum | op1
+        elif sum > MAX:
+            print("bigger")
+        sum = op1 & op2
+        return {'oper': '&', 'op2': op2, 'sum': sum}
+
+    def bitxor(self, op1, op2):
+        if op1 > 0:
+            op2 = int(math.fabs(op2))
+        else:
+            op2 = - int(math.fabs(op2))
+        sum = op2 ^ op1
+        return {'oper': '^', 'op2': op2, 'sum': sum}
+
+    def debug(self):
+        last = self.operands[0]
+        for i in range(self.num_operands -1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i+1])
+            print("%d. %d %s %d == %d" % (i, last, result["oper"], result["op2"], result["sum"]))
+            self.operands[i+1] = result["op2"]
+            last = result["sum"]
+
+    def printJS(self):
+        last = self.operands[0]
+        print("%s%d" % ("(" * (self.num_operands -1), last), end='')
+        for i in range(self.num_operands -1):
+            result = self.operators_list[self.operators[i]](last, self.operands[i+1])
+            print(" %s %d)" % (result["oper"], result["op2"]), end='')
+            self.operands[i+1] = result["op2"]
+            last = result["sum"]
+        print(" == %d" % last,)
+
+
+
+# gs = GenSequence(15)
+# gs.debug()
+# gs.printJS()
+def generate_binary_tests(options):
+    if not os.path.isdir(options.output):
+        os.makedirs(options.output)
+    SEED = options.seed[0]
+    random.seed(SEED)
+    counter = 0
+    file_count = 1
+    new_file_begin = True
+    bitcalc = False
+    arithcalc = False
+    for i in range(len(options.op)):
```

Comment:
Same as above. Use `for op in options.op`

Reply:
Fixed

Reply:
Did you really refactor this code? Use the element itself instead of the index (as I see you have never reused the index other than calculating the element)

Reply:
It seems unresolved.